managed/src/test/java/com/yugabyte/yw/controllers/CustomerConfigControllerTest.java

```diff
@@ -169,4 +169,42 @@ public void testDeleteInUseStorageConfig() {
     assertEquals(0, CustomerConfig.getAll(defaultCustomer.uuid).size());
     assertAuditEntry(1, defaultCustomer.uuid);
   }
+
+  @Test
+  public void testEditInUseStorageConfig() {
+    ObjectNode bodyJson = Json.newObject();
+    JsonNode data = Json.parse("{\"foo\":\"bar\"}");
+    bodyJson.put("name", "test1");
+    bodyJson.set("data", data);
+    bodyJson.put("type", "STORAGE");
+    UUID configUUID = ModelFactory.createS3StorageConfig(defaultCustomer).configUUID;
+    Backup backup = ModelFactory.createBackup(defaultCustomer.uuid, UUID.randomUUID(),
+                                              configUUID);
+    String url = "/api/customers/" + defaultCustomer.uuid + "/configs/" + configUUID;
+    Result result = FakeApiHelper.doRequestWithAuthTokenAndBody("PUT", url,
+        defaultUser.createAuthToken(), bodyJson);
+    assertOk(result);
+    backup.delete();
+    result = FakeApiHelper.doRequestWithAuthTokenAndBody("PUT", url,
+        defaultUser.createAuthToken(), bodyJson);
+    assertOk(result);
+    assertEquals(data, CustomerConfig.get(configUUID).data);
+  }
+
```

Comment:
Required to add more test cases. 
1. testcase to edit not used config. It should allow all fields to edit.
2. testcase to edit in the used config should config with limited fields.
3. In used config with all field edit. API should give an error.
4. Invalid Data, Content, and Invalid configured. you can add 1 test case for each or add one testcase and cover all 3 senario.

Reply:
@jaydeepkumara for both used and unused configs, editable fields are the same I believe. ( at least the current implementation of UI behaves like that. @nishantSharma459 confirm )

Reply:
ok.